core/types/transaction_signing.go

```diff
@@ -108,12 +108,7 @@ func SignTx(tx *Transaction, s Signer, prv *ecdsa.PrivateKey) (*Transaction, err
 // SignNewTx creates a transaction and signs it.
 func SignNewTx(prv *ecdsa.PrivateKey, s Signer, txdata TxData) (*Transaction, error) {
 	tx := NewTx(txdata)
-	h := s.Hash(tx)
-	sig, err := crypto.Sign(h[:], prv)
-	if err != nil {
-		return nil, err
-	}
-	return tx.WithSignature(s, sig)
+	return SignTx(tx, s, prv)
```

Comment:
Please inline `NewTx(txdata)` in place of `tx`, the line's short anyway and less code, if we're cleaning up either way.

Reply:
done.